pkg/edit/edit.go

```diff
@@ -0,0 +1,108 @@
+package edit
+
+import (
+	"fmt"
+	"os"
+	"path"
+	"strings"
+
+	errors "github.com/zgalor/weberr"
+)
+
+type machineType struct {
+	id      string
+	ccsOnly string
+	enabled string
+}
+
+type cloudRegion struct {
+	id       string
+	enabled  string
+	govcloud string
+	ccsOnly  string
+}
+
+type ConstraintMap struct {
+	entireFile   []string
+	machineTypes []machineType
+	cloudRegions []cloudRegion
+	Path         string
+}
+
+func (c *ConstraintMap) ReadConstraints(env string, appInterfaceDir string) error {
+	c.Path = path.Join(appInterfaceDir,
+		"resources/services/ocm/"+env+"/cloud-resource-constraints.configmap.yaml")
+	yamlFile, err := os.ReadFile(c.Path)
+	if err != nil {
+		return errors.Errorf("File read err: '%v'", err)
+	}
+
+	for _, line := range strings.Split(strings.TrimSuffix(string(yamlFile), "\n"), "\n") {
+		c.entireFile = append(c.entireFile, line)
+	}
+
+	return nil
+}
+
+func (c *ConstraintMap) EditConstraint(id string, ccsOnly *bool, enabled *bool, govcloud *bool) error {
+	edit := false
+	changeMade := false
+	for i, line := range c.entireFile {
+		if strings.Contains(line, "id:") {
+			edit = false
+			if strings.Contains(line, id) {
+				edit = true
+			}
+		}
+		if edit {
+			if ccsOnly != nil && strings.Contains(line, "ccs_only") {
+				s := "false"
+				if *ccsOnly {
+					s = "true"
+				}
+				changeMade = true
+				c.entireFile[i] = strings.Split(line, ": ")[0] + ": " + s
+			} else if enabled != nil && strings.Contains(line, "enabled") {
+				s := "false"
+				if *enabled {
+					s = "true"
+				}
+				changeMade = true
+				c.entireFile[i] = strings.Split(line, ": ")[0] + ": " + s
+			} else if govcloud != nil && strings.Contains(line, "govcloud") {
+				s := "false"
+				if *govcloud {
+					s = "true"
+				}
+				changeMade = true
+				c.entireFile[i] = strings.Split(line, ": ")[0] + ": " + s
+			}
+		}
+	}
+	if !changeMade {
+		fmt.Printf("\nNo changes were made, writing same data back into constraintmap " +
+			"(use '--help' to see what you can change via flags)")
+	}
+	return nil
+}
+
+func (c *ConstraintMap) SaveFile() error {
+
+	if err := os.Truncate(c.Path, 0); err != nil {
```

Comment:
`os.WriteFile` will do the truncation for you according to the go docs: https://pkg.go.dev/os#WriteFile 

The trade-off is that you run the risk of partial file writes if the command fails half-way through. Your call 👍

Reply:
Interesting, didn't know about that. truncate has been used in other places in this CLI so I think i'll keep for consistency, but may refactor later :slightly_smiling_face:

Reply:
Sure thing @hunterkepley . Good call on the consistency side of things.

Reply:
Capture it in a ticket, and refactor later to keep the consistency, would be a good first issue for someone wanting to get started on this